src/routes/PrivateRoute.js

```diff
@@ -0,0 +1,8 @@
+import React from 'react';
+import { useSelector } from 'react-redux';
+import { Navigate } from 'react-router-dom';
+
+export default function PrivateRoute({ children }) {
+    const { data } = useSelector((state) => state.api);
+    return (Array.isArray(data)) ?  <Navigate to="/" replace={true}/> :children
```

Comment:
check for the length as well as in case some case it might be an empty array as well which will break the logic

Reply:
done